CDCgov/prime-reportstream
# `<IACardGridTemplate>` now optionally configurable with sections

This PR adds new functionality for `IACardGridTemplate` -- it now no longer only takes a `ContentDirectory[]`. You can, instead, pass in a map of directory arrays where the map keys act as the title, and the values act as the cards to render.

For an example (commented code), see `content/resources/index.ts`!

Test Steps:
1. Ensure no IA pages are broken from this
2. To test sections, uncomment the sample in `content/resources/index.ts` then go to `pages/resources/ResourceCardGrid.tsx` and swap the value on line `17` to `resourcesContentMap` (requires you to import it now that it's uncommented in the other file). This only renders ONE section, but it shows what a section looks like.

If you tinker enough, you can try creating your own section below it! Though, take this as my proof that it works 😄
<img width="1436" alt="Screen Shot 2022-08-18 at 4 38 36 PM" src="https://user-images.githubusercontent.com/4022304/185495777-c9297ab0-244e-4c34-8898-e6613f019667.png">


## Changes
- `IACardGridTemplate` takes new `ContentMap` and names sections of cards
- Enumerates the page names of `settings` and `resources` inside their index files (cleanup task)
- Updates snapshot for a preexisting test

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #6278 

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
